lib/mayaUsd/render/vp2RenderDelegate/proxyRenderDelegate.h

```diff
@@ -295,13 +295,49 @@ class ProxyRenderDelegate : public MHWRender::MPxSubSceneOverride
     //! A collection of Rprims to prepare render data for specified reprs
     std::unique_ptr<HdRprimCollection> _defaultCollection;
 
-    //! The render tag version used the last time render tags were updated
-    unsigned int _renderTagVersion { 0 }; // initialized to 1 in HdChangeTracker, so we'll always
-                                          // have an invalid version the first update.
-#ifdef ENABLE_RENDERTAG_VISIBILITY_WORKAROUND
-    unsigned int _visibilityVersion { 0 }; // initialized to 1 in HdChangeTracker.
-#endif
-    bool _taskRenderTagsValid {
+    // Version Id values saved from the last time we queried the HdChangeTracker.
+    // These values are initialized to 1 in HdChangeTracker
+    struct HdChangeTrackerVersions
+    {
+        //! The render tag version the last time _Execute was called
+        unsigned int _renderTag { 0 };
+        //! The visibility change count the last time _Execute was called
+        unsigned int _visibility { 0 };
+        //! The combined instancer index version and instance index change count the last time
+        //! _Execute was called
+        unsigned int _instanceIndex { 0 };
+
+        void sync(const HdChangeTracker& tracker)
+        {
+            _renderTag = tracker.GetRenderTagVersion();
+            _visibility = tracker.GetVisibilityChangeCount();
+            _instanceIndex = tracker.GetInstancerIndexVersion();
```

Comment:
GetInstanceIndexVersion doesn't capture all the workflows we care about, but it captures some of them. So at least some things work correctly with this updated version of the code.